lib/common/utils.ts

```diff
@@ -148,10 +148,45 @@ export function patchProperty(obj: any, prop: string) {
   Object.defineProperty(obj, prop, desc);
 }
 
+function getEventNames(obj: any, eventNames: string[]) {
+  const onProperties = [];
+  for (const prop in obj) {
+    if (prop.substr(0, 2) == 'on' && prop.length > 2) {
+      onProperties.push(prop.substr(2));
+    }
+  }
+  onProperties.sort();
+  eventNames.sort();
+
+  const validEventNames: string[] = [];
+  const onPropertiesLength = onProperties.length;
+  const eventNamesLength = eventNames.length;
+
+  let i = 0;
+  let j = 0;
+
+  while (i < onPropertiesLength && j < eventNamesLength) {
+    const prop = onProperties[i];
+    const eventName = eventNames[j];
+    const compareResult = prop.localeCompare(eventName);
+    if (compareResult === 0) {
+      i++;
+      j++;
+      validEventNames.push(eventName);
+    } else if (compareResult < 0) {
+      i++;
+    } else {
+      j++;
+    }
+  }
+  return validEventNames;
+}
+
 export function patchOnProperties(obj: any, properties: string[]) {
   if (properties) {
-    for (let i = 0; i < properties.length; i++) {
-      patchProperty(obj, 'on' + properties[i]);
+    const validEventNames = getEventNames(obj, properties);
```

Comment:
@mhevery, yes, in my first commit, my goal is to make a fixed set, but it seems sometimes patch non exist property may cause errors, I will do more research and describe it more clear here.